CONTRIBUTING.md

```diff
@@ -0,0 +1,77 @@
+* **Bicep Language Server** (`src/Bicep.LangServer`): the LanguageServer used by the VSCode extension for parsing and providing information about a Bicep file.
+* **Bicep Core** (`src/Bicep.Core`): the library containing the majority of the Biceo compiler code.
+* **Bicep VSCode Extension** (`src/vscode-bicep`): the VSCode extension itself. This is mostly a thin wrapper around the Bicep Language Server.
+* **Playground** (`src/playground`): the web-based playground hosted at `https://aka.ms/bicepdemo`.
+* A number of different test suites.
+
+### Running the tests
+* You can use the following command to run the full Bicep test suite:
+    * `dotnet test`
+
+### Running the Bicep VSCode extension
+* On the first run, you'll need to ensure you have installed all the npm packages required by the Bicep VSCode extension with the following:
+    * `cd src/bicep-vscode`
+    * `npm i`
+* In the [VSCode Run View](https://code.visualstudio.com/Docs/editor/debugging), select the "Bicep VSCode Extension" task, and press the "Start" button. This will launch a new VSCode window with the Bicep extension and LanguageServer containing your changes.
+* If you want the ability to put breakpoints and step through the C# code, you can also use the "Attach" run configuration once the extension host has launched, and select the Bicep LanguageServer process.
+
+### Running the Bicep CLI
+* In the [VSCode Run View](https://code.visualstudio.com/Docs/editor/debugging), select the "Bicep Cli" task, and press the "Start" button. This will build and run the Bicep CLI and allow you to step through the code.
+* Note that usually you will want to pass your own custom arguments to the Bicep CLI. This can be done by modifying the `launch.json` configuration to add arguments to the "args" array for the "Bicep Cli" task. 
+
+### Running the Playground
+* On the first run, you'll need to ensure you have installed all the npm packages required by the Bicep Playground with the following:
+    * `cd src/playground`
+    * `npm i`
+* In the [VSCode Run View](https://code.visualstudio.com/Docs/editor/debugging), select the "Bicep Playground" task, and press the "Start" button. This will launch a browser window with the Playground containing your changes.
+
+## Pull Requests
+TBD. Notes:
+* Add info about "good first issue" issue tag.
+* Create issue first? (ensure the PR is addressing a specific issue)
+* Always best to speak to us first if it's a significant change to the design.
+
+## Example Files
+If you'd like to contribute example `.bicep` files that showcase abilities of the language:
+* Create an appropriately-named directory inside `docs/examples`.
+* Include your file named `main.bicep`.
+* Compile the file using the Bicep CLI, and include the compiled `main.json` with your check-in.
+* Our Pull Request validation will ensure that the `main.bicep` file can be compiled successfully without errors, and that the generated `main.json` file matches the one being checked in.
```

Comment:
Maybe add a reminder that dotnet test from root will validate locally as well?

Reply:
+1, I wish I had known this earlier :)